criu/arch/mips/vdso-pie.c

```diff
@@ -0,0 +1,76 @@
+#include <unistd.h>
+
+#include "asm/types.h"
+
+#include <compel/plugins/std/string.h>
+#include <compel/plugins/std/syscall.h>
+#include "parasite-vdso.h"
+#include "log.h"
+#include "common/bug.h"
+
+#ifdef LOG_PREFIX
+# undef LOG_PREFIX
+#endif
+#define LOG_PREFIX "vdso: "
+
+static void insert_trampoline32(uintptr_t from, uintptr_t to)
+{
+	struct {
+		u8	movl;
+		u32	imm32;
+		u16	jmp_eax;
+		u32	guards;
+	} __packed jmp = {
+		.movl		= 0xb8,
+		.imm32		= (uint32_t)to,
+		.jmp_eax	= 0xe0ff,
+		.guards		= 0xcccccccc,
+	};
+
+	memcpy((void *)from, &jmp, sizeof(jmp));
+}
+
+static void insert_trampoline64(uintptr_t from, uintptr_t to)
+{
+	struct {
+		u16	movabs;
+		u64	imm64;
+		u16	jmp_rax;
+		u32	guards;
+	} __packed jmp = {
+		.movabs		= 0xb848,
+		.imm64		= to,
+		.jmp_rax	= 0xe0ff,
+		.guards		= 0xcccccccc,
+	};
```

Comment:
you need to adjust the code for mips64 assembly here.
For more information, what's this about:
https://criu.org/Vdso